sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3077,12 +3077,16 @@ object functions {
   //////////////////////////////////////////////////////////////////////////////////////////////
 
   /**
-   * Returns null if the array is null, true if the array contains `value`, and false otherwise.
+   * Returns null if the array is null, true if the array contains `value` or the content of
+   * `value` if it is of type Column, and false otherwise.
    * @group collection_funcs
    * @since 1.5.0
    */
   def array_contains(column: Column, value: Any): Column = withExpr {
-    ArrayContains(column.expr, Literal(value))
```

Comment:
Other similar expressions are `ArrayPosition`, `ElementAt`, `ArrayRemove`.

Reply:
Oh, I just found @maropu has pointed it out in https://github.com/apache/spark/pull/21581#issuecomment-397928010.